lib/spack/spack/package.py

```diff
@@ -2142,15 +2142,33 @@ def fetch_remote_versions(self):
             tty.die("Package.fetch_versions couldn't connect to:", e.url,
                     e.message)
 
+    @property
+    def rpath_deps(self):
+        """Return immediate or transitive RPATHs depending on the package."""
+        if self.transitive_rpaths:
+            return [d for d in self.spec.traverse(root=False,
+                                                  deptype=('link'))]
+        else:
+            return self.spec.dependencies(deptype='link')
+
     @property
     def rpath(self):
         """Get the rpath this package links with, as a list of paths."""
-        rpaths = [self.prefix.lib, self.prefix.lib64]
-        deps = self.spec.dependencies(deptype='link')
-        rpaths.extend(d.prefix.lib for d in deps
```

Comment:
@scheibelp it's hard to say who calls `Package::rpath()`, but the current code is also wrong. Thus the changes are relevant.